### PR TITLE
Add Threshold to Energy Detector Cover

### PR DESCRIPTION
**Pull request by MuXiu1997**
Closed https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/9381

image

### CODE REVIEW

File: src/main/java/gregtech/common/covers/GT_Cover_EUMeter.java [hunk 1]
```diff
@@ -1,282 +1,502 @@
...
+            GT_Utility.trans("306", "EU stored"),
+            GT_Utility.trans("261", "Electricity Storage(Including Batteries)"),
+            (te) -> {
+                long stored = te.getStoredEU();
+                if (te instanceof IGregTechTileEntity) {
+                    IGregTechTileEntity tTileEntity = (IGregTechTileEntity) te;
+                    IMetaTileEntity mTileEntity = tTileEntity.getMetaTileEntity();
+                    if (mTileEntity instanceof GT_MetaTileEntity_BasicBatteryBuffer) {
+                        GT_MetaTileEntity_BasicBatteryBuffer buffer = (GT_MetaTileEntity_BasicBatteryBuffer) mTileEntity;
+                        if (buffer.mInventory != null) {
```
Comment (repo-alt) [inline src/main/java/gregtech/common/covers/GT_Cover_EUMeter.java:308]: Actually this somewhat duplicates GT_MetaTileEntity_BasicBatteryBuffer.getStoredEnergy

Comment (MuXiu1997) [reply src/main/java/gregtech/common/covers/GT_Cover_EUMeter.java:308]: Thanks, I only wrote it based on the original code of this file before

File: src/main/java/gregtech/common/covers/GT_Cover_EUMeter.java [hunk 2]
```diff
@@ -1,282 +1,466 @@
...
 
     @Override
-    public Object getClientGUI(byte aSide, int aCoverID, int coverData, ICoverable aTileEntity)  {
+    public Object getClientGUIImpl(byte aSide, int aCoverID, EUMeterData coverData, ICoverable aTileEntity, EntityPlayer aPlayer, World aWorld) {
         return new GUI(aSide, aCoverID, coverData, aTileEntity);
     }
+    //endregion
+
+    public static class EUMeterData implements ISerializableObject {
+        private byte type;
```
Comment (Glease) [inline src/main/java/gregtech/common/covers/GT_Cover_EUMeter.java:187]: A better choice is to just store EnergyType in the field. No need to look it up whenever you need it.

Comment (MuXiu1997) [reply src/main/java/gregtech/common/covers/GT_Cover_EUMeter.java]: Ok

**Review by Glease — APPROVED** [review]

**Pull request merged**